sdk/resource/doc.go

```diff
@@ -25,4 +25,6 @@
 // OTEL_RESOURCE_ATTRIBUTES the FromEnv Detector can be used. It will interpret
 // the value as a list of comma delimited key/value pairs
 // (e.g. `<key1>=<value1>,<key2>=<value2>,...`).
+//
+// Telemetry Stability: Unstable.
```

Comment:
I think this needs to explained further. I expect users will read this and not understand what it is trying to say.

Maybe something like:

```suggestion
//  While this package provides a stable API, the attributes for a detected resource may change.
```

Reply:
Maybe as an additional sentence?

```suggestion
// Telemetry Stability: Unstable.
// The attributes for a detected resource may change in future releases.
```

Reply:
There are two things that worry me with the addition of "Telemetry Stability: Unstable."

1. This package doesn't produce telemetry. It is the attributes of the resource that are unstable.
2. Saying something is unstable is going to confuse the user into thinking the package API is unstable. I can see a user thinking "Telemetry Stability" refers to OpenTelemetry stability and then seeing "unstable" to mean this whole thing is unstable.

If the goal of the documentation is help explain things to the user, this addition doesn't seem to be doing that.

Reply:
I see your point and I agree with you. Changing to your suggestion.

```suggestion
// While this package provides a stable API,
// the attributes for a detected resource may change.
```

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/4080/commits/d065c4344f3601ce89957e920f901b53fb3da33e